src/pocketdb/repositories/web/ExplorerRepository.cpp

```diff
@@ -382,4 +382,46 @@ namespace PocketDb {
         });
     }
 
+    UniValue ExplorerRepository::GetBalanceHistory(const vector<string>& addresses, int topHeight, int count)
+    {
+        UniValue result(UniValue::VARR);
+
+        TryTransactionStep(__func__, [&]()
+        {
+            auto stmt = SetupSqlStatement(R"sql(
+                select b.Height, sum(b.Value)Amount
+                from Balances b indexed by Balances_Height
+                where b.AddressHash in ( )sql" + join(vector<string>(addresses.size(), "?"), ",") + R"sql( )
+                  and b.Height <= ?
+                group by b.Height
```

Comment:
The total amount for all addresses? Not separately for each address?

Reply:
Yes, the call is scheduled for use in schedules, so the overall balance. Detailing is already a separate issue.